src/overlays/actors/ovl_En_Ma1/z_en_ma1.c

```diff
@@ -19,45 +36,410 @@ const ActorInit En_Ma1_InitVars = {
+    if (gSaveContext.eventChkInf[1] & 0x40) {
+        if (player->stateFlags2 & 0x1000000) {
+            player->stateFlags2 |= 0x2000000;
+            player->unk_6A8 = &this->actor;
+            this->actor.textId = 0x2061;
+            func_8010B680(globalCtx, this->actor.textId, 0);
+            this->unk_1E8.unk_00 = 1;
+            this->actor.flags |= 0x10000;
+            this->actionFunc = (ActorFunc)func_80AA106C;
+        } else if (this->actor.xzDistanceFromLink < 30.0f + (f32)this->collider.dim.radius) {
+            player->stateFlags2 |= 0x800000;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/EnMa1_Init.s")
+void func_80AA106C(EnMa1* this, GlobalContext* globalCtx) {
+    PLAYER->stateFlags2 |= 0x800000;
+    if (this->unk_1E8.unk_00 == 2) {
+        func_800ED858(2);
+        func_8010BD58(globalCtx, 9);
+        this->actor.flags &= ~0x10000;
+        this->actionFunc = (ActorFunc)func_80AA10EC;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/EnMa1_Destroy.s")
+void func_80AA10EC(EnMa1* this, GlobalContext* globalCtx) {
+    PLAYER->stateFlags2 |= 0x800000;
+    if (func_8010BDBC(&globalCtx->msgCtx) == 7) {
+        func_8010BD58(globalCtx, 0x16);
+        this->actionFunc = (ActorFunc)func_80AA1150;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0D88.s")
+void func_80AA1150(EnMa1* this, GlobalContext* globalCtx) {
+    PLAYER->stateFlags2 |= 0x800000;
+    if (globalCtx->msgCtx.unk_E3EE == 3) {
+        globalCtx->nextEntranceIndex = 0x157;
+        gSaveContext.nextCutsceneIndex = 0xFFF1;
+        globalCtx->fadeTransition = 42;
+        globalCtx->sceneLoadFlag = 0x14;
+        this->actionFunc = (ActorFunc)func_80AA11C8;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0EA0.s")
+void func_80AA11C8(EnMa1* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0EFC.s")
+void EnMa1_Update(EnMa1* this, GlobalContext* globalCtx) {
+    ColliderCylinder* collider = &this->collider;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0F44.s")
+    Collider_CylinderUpdate(&this->actor, collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &collider->base);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80AA0A0C(this);
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc != (ActorFunc)func_80AA11C8) {
+        func_800343CC(globalCtx, &this->actor, &this->unk_1E8.unk_00, (f32)this->collider.dim.radius + 30.0f,
+                      EnMa1_GetText, func_80AA0778);
+    }
+    func_80AA0B74(this);
+    func_80AA0AF4(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA106C.s")
+s32 func_80AA12BC(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, EnMa1* this) {
```

Comment:
EnMa1_OverrideLimbDraw. this and the function below are common callbacks for SkelAnime_DrawSV, so we can go with this naming convention consistently